pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -60,41 +87,100 @@ func (m *cgroupManagerImpl) Destroy(cgroupConfig *CgroupConfig) error {
+		}
+		if err := sys.Set(cgroupConfig.Paths[sys.Name()], cgroupConfig); err != nil {
+			return fmt.Errorf("Failed to set config for subsystem %v: %v", sys.Name, err)
+		}
+	}
+	if cgroupConfig.Paths["cpu"] != "" {
+		if err := cgroupfs.CheckCpushares(cgroupConfig.Paths["cpu"], cgroupConfig.Resources.CpuShares); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
 // Update updates the cgroup with the specified Cgroup Configuration
 func (m *cgroupManagerImpl) Update(cgroupConfig *CgroupConfig) error {
 	//cgroup name
 	name := cgroupConfig.Name
 
-	// get the fscgroup Manager with the specified cgroup configuration
-	fsCgroupManager, err := getLibcontainerCgroupManager(cgroupConfig, m.subsystems)
-	if err != nil {
-		return fmt.Errorf("Failed to update cgroup for %v : %v", name, err)
+	// Get map of all cgroup paths on the system for the particular cgroup
+	cgroupPaths := make(map[string]string, len(m.subsystems.mountPoints))
+	for key, val := range m.subsystems.mountPoints {
+		cgroupPaths[key] = path.Join(val, name)
 	}
-	// get config object for passing to Set()
-	config := &libcontainerconfigs.Config{
-		Cgroups: fsCgroupManager.Cgroups,
+	// Extract the cgroup resource parameters
+	resourceConfig := cgroupConfig.ResourceParameters
+	resources := &libcontainerconfigs.Resources{}
+	if resourceConfig.Memory != nil {
+		resources.Memory = *resourceConfig.Memory
 	}
-
-	// Update cgroup configuration using libcontainers Managers Set() method
-	if err := fsCgroupManager.Set(config); err != nil {
-		return fmt.Errorf("Failed to update cgroup for %v: %v", name, err)
+	if resourceConfig.CpuShares != nil {
+		resources.CpuShares = *resourceConfig.CpuShares
+	}
+	if resourceConfig.CpuQuota != nil {
+		resources.CpuQuota = *resourceConfig.CpuQuota
+	}
+	// Initialize libcontainer's cgroup config
+	libcontainerCgroupConfig := &libcontainerconfigs.Cgroup{
+		Name:      path.Base(name),
+		Parent:    path.Dir(name),
+		Resources: resources,
+		Paths:     cgroupPaths,
+	}
+	if err := setSupportedSubsytems(libcontainerCgroupConfig); err != nil {
+		return fmt.Errorf("Failed to set supported cgroup subsystems for cgroup %v: %v", name, err)
 	}
 	return nil
 }
 
 // Create creates the specified cgroup
 func (m *cgroupManagerImpl) Create(cgroupConfig *CgroupConfig) error {
-	//cgroup name
+	// get cgroup name
 	name := cgroupConfig.Name
 
-	// get the fscgroup Manager with the specified cgroup configuration
-	fsCgroupManager, err := getLibcontainerCgroupManager(cgroupConfig, m.subsystems)
-	if err != nil {
-		return fmt.Errorf("Failed to create cgroup for %v : %v", name, err)
+	// Get map of all cgroup paths on the system for the particular cgroup
+	cgroupPaths := make(map[string]string, len(m.subsystems.mountPoints))
+	for key, val := range m.subsystems.mountPoints {
+		cgroupPaths[key] = path.Join(val, name)
+	}
+	// Only create paths which don't already exist
+	pathsToCreate := make(map[string]string)
+	for key, val := range cgroupPaths {
```

Comment:
you only use `cgroupsPaths` here so I think this can all be in the original loop on line 162:

```go
pathsToCreate := make(map[string]string)
for key, val := range m.subsystems.mountPoints {
    pathsToCreate[key] = path.Join(val, name)
}
```